lib/plugins/aws/deploy/lib/mergeIamTemplates.js

```diff
@@ -11,131 +11,141 @@ module.exports = {
   },
 
   merge() {
+    // resolve early if no functions are provided
     if (!this.serverless.service.getAllFunctions().length) {
       return BbPromise.resolve();
     }
 
-    let anyFunctionHasNoRole = false;
```

Comment:
@pmuens @eahefnawy @mthenw I updated this logic as it seemed to be broken, or my brain didn't compute 😊